spec/unit/agent_spec.rb

```diff
@@ -188,7 +188,6 @@ def controlled_run(&block)
 
         allow(Kernel).to receive(:fork)
         allow(Process).to receive(:waitpid2).and_return([123, double('process::status', :exitstatus => 0)])
-        allow(@agent).to receive(:exit)
       end
```

Comment:
There's a lot of overstubbing in puppet and this is a good example. We're trying to test that fork/waitpid/exit/etc work as expected yet all of those calls are stubbed out. In cases like this, I generally rewrite the test to no longer stub in one commit. Then add another commit with your changes (6b085cc6eb00f068e57127fcd9de7ab408c9d829). That way it's clear what behaviors are being changed in the second commit. For example, it'll be clear that we preserve the exit code of the forked child rather than returning `-1`.

Reply:
Ah yeah, I was thinking we rip out all of the `Kernel.fork`, `Process.waitpid2`, etc stubbing in the "when should_fork is true" describe. The tests aren't very useful because everything is so stubbed out. But it's fine if you don't want to change that.

Reply:
I order to keep the tests usefulness, beside removing overstubbing I also removed `SystemExit` rescuing